lib/kalibro_client/entities/processor/metric_collector_details.rb

```diff
@@ -33,8 +33,7 @@ def find_metric_by_name(name)
         end
 
         def find_metric_by_code(metric_code)
-          metric = self.supported_metrics.find {|code, metric| metric.code == metric_code}
-          metric.nil? ? nil : metric.last
+          @supported_metrics[metric_code]
```

Comment:
Could we create another hash `supported_metrics_by_name` to be used in `find_metric_by_name`?